### PR TITLE
Add `.memoized?` method

### PR DESCRIPTION
**Pull request by AlexWayfer**
To check (externally) if method memoized or not.

### CODE REVIEW

File: spec/memery_spec.rb [hunk 1]
```diff
@@ -254,4 +265,52 @@ class E
       end
     end
   end
+
+  describe ".memoized?" do
+    subject { object.memoized?(method_name) }
+
+    shared_examples "works correct" do
```
Comment (tycooon) [inline spec/memery_spec.rb]: Should be `works correctly`.

**Comment by tycooon** [comment]
Please add new method to the Readme.

File: spec/memery_spec.rb (hunk 1 above)
Comment (AlexWayfer) [reply spec/memery_spec.rb]: Yes, sorry. Fixed.

**Comment by AlexWayfer** [comment]
> Please add new method to the Readme.

Yes, done.

File: lib/memery.rb [hunk 2]
```diff
@@ -31,6 +31,13 @@ def memoize(method_name, condition: nil, ttl: nil)
       define_memoized_method!(method_name, condition: condition, ttl: ttl)
     end
 
+    def memoized?(method_name)
+      return faslse unless defined?(@_memery_module)
```
Comment (tycooon) [inline lib/memery.rb]: There is a typo so I guess we should test it.

Comment (AlexWayfer) [reply lib/memery.rb]: I agree. Sorry. Added.

**Comment by tycooon** [comment]
Please check the coverage, it looks like there is some dead code in specs: https://coveralls.io/builds/25906116

**Comment by AlexWayfer** [comment]
> Please check the coverage, it looks like there is some dead code in specs

I think we should filter `/spec` from coverage, but OK, I've removed the body of unused method.

**Comment by tycooon** [comment]
> > Please check the coverage, it looks like there is some dead code in specs
> 
> I think we should filter `/spec` from coverage, but OK, I've removed the body of unused method.

It still shows coverage dercrease. Why do we need that the code that is never called in specs? 🤔

**Comment by AlexWayfer** [comment]
> It still shows coverage decrease. Why do we need the code that is never called in specs? thinking

OK, another one, I'm sorry. Removed.

File: lib/memery.rb [hunk 3]
```diff
@@ -31,6 +31,13 @@ def memoize(method_name, condition: nil, ttl: nil)
       define_memoized_method!(method_name, condition: condition, ttl: ttl)
     end
 
+    def memoized?(method_name)
+      return false unless defined?(@_memery_module)
+
+      @_memery_module.method_defined?(method_name) ||
```
Comment (tycooon) [inline lib/memery.rb]: What if method is protected? I think it’s better to use Memery.method_visibility in this case.

Comment (AlexWayfer) [reply lib/memery.rb]: > What if method is protected?

`true`: https://ruby-doc.org/core-2.6.4/Module.html#method-i-method_defined-3F

> Public and protected methods are matched.

**Pull request merged**